### ISSUE TITLE
replication models - should they become service classes? Or ...?

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/preservation_catalog#2190]
This is a question in that there is a choice to be made.

PR #2188 moved app/models/audit_results.rb to app/services/audit/results.rb.

It makes me wonder if app/models/replication/   DruidVersionZip and DruidVersionZipPart should also be service classes, to be consistent.

Or if I should not have merged PR #2188 and we should have instead moved AuditResults into app/models/audit.

I don't care much which way we go, but the inconsistency bothers me.

(tagging @mjgiarlo who was very focused on PR #2188 having followed what the ticket said.  I wrote the ticket and I'm imperfect.)

**Comment by mjgiarlo** [comment]
@sul-dlss/infrastructure-team Typing this out before anyone sinks too much time on this issue, unpacking why I was hesitant to tackle this as part of #2188.

My take is that all of the classes we're talking about here are a mess of tangled concerns, and trying to fit any of them neatly into "models" or "service classes" may not be worth the effort. (And closing this ticket and replacing it with a new "make better abstractions!" ticket is of limited value, too; maybe more noise than anything else?)

Why do I say this? Look at the public interface of these classes. They contain between 8 and 19 public methods, not even counting attrs or delegations. Personally, I think we may be in diminishing returns territory with prescat refactors so I'd not be inclined to run with this one. But! If anyone else is super bothered by the current organization and wants to refactor here, I wouldn't discourage them.

As for `AuditResults`, I appreciate that moving that class into `app/services/` gets it closer to some its main collaborators, i.e., the audit results reporters.

As for the two classes in `app/models/replication/`, they don't strike me as terribly service-y. They smell a little bit like value objects, but they have a heck of a lot more responsibilities than I'd expect in a value object. It's OK to have POROs in `app/models/` so...

Long story short, my own sense is that the current organization might make the most sense, but I welcome other POVs!

**Comment by ndushay** [comment]
I'm closing this;  I can live with the inconsistencies.
